std/traits.d

```diff
@@ -6967,7 +6966,7 @@ enum bool isSIMDVector(T) = is(T : __vector(V[N]), V, size_t N);
 /**
  * Detect whether type `T` is a pointer.
  */
-enum bool isPointer(T) = is(T == U*, U) && !isAggregateType!T;
+enum bool isPointer(T) = is(T == U*, U) && __traits(isScalar, T);
```

Comment:
It's not clear to me that `is(T == U*, U)` could ever be true for an aggregate type `T` but I kept a check just in case there's some nuance I'm overlooking.

Reply:
Seems like a workaround for an old bug:

```
struct S
{
    int* ptr;
    alias ptr this;
}

pragma(msg, is(S == T*, T));
```
```
2.063   to 2.083.1: Success with output: true
Since      2.084.1: Success with output: false
```